### ISSUE TITLE
mrjob on Hadoop 2.7.1 returned non-zero exit status 256

### ISSUE THREAD

**Issue by wsshopping** [Yelp/mrjob#1092]
HI, i am a newer about mrjob and hadoop, after i build my hadoop cluster, i try to use mrjob submit the job to hadoop,
but unfortunatly, it failed with the error "returned non-zero exit status 256".more details as follow:

my enviroment:
hadoop:2.7.1
python:2.7.9

#1.this is my example:

```python
"""The classic MapReduce job: count the frequency of words.
"""
#! /usr/bin/python

from mrjob.job import MRJob
import re

WORD_RE = re.compile(r"[\w']+")


class MRWordFreqCount(MRJob):

    def mapper(self, _, line):
        for word in WORD_RE.findall(line):
            yield (word.lower(), 1)

    def combiner(self, word, counts):
        yield (word, sum(counts))

    def reducer(self, word, counts):
        yield (word, sum(counts))


if __name__ == '__main__':
     MRWordFreqCount.run()
```

#2. and i use this command:
```python
python test.py -r hadoop  --python-bin=/root/.pyenv/versions/2.7.9/bin/python   ./pg20417.txt  
```

#3. this is the result what i got:
```python
HADOOP:  map 100% reduce 100%
HADOOP: Job job_1437555855039_0008 failed with state FAILED due to: Task failed task_1437555855039_0008_m_000001
HADOOP: Job failed as tasks failed. failedMaps:1 failedReduces:0
HADOOP: 
HADOOP: Counters: 13
HADOOP:     Job Counters 
HADOOP: 		Failed map tasks=7
HADOOP: 		Killed map tasks=1
HADOOP: 		Launched map tasks=8
HADOOP: 		Other local map tasks=6
HADOOP: 		Data-local map tasks=2
HADOOP: 		Total time spent by all maps in occupied slots (ms)=136520
HADOOP: 		Total time spent by all reduces in occupied slots (ms)=0
HADOOP: 		Total time spent by all map tasks (ms)=17065
HADOOP: 		Total vcore-seconds taken by all map tasks=17065
HADOOP: 		Total megabyte-seconds taken by all map tasks=17474560
HADOOP: 	Map-Reduce Framework
HADOOP: 		CPU time spent (ms)=0
HADOOP: 		Physical memory (bytes) snapshot=0
HADOOP: 		Virtual memory (bytes) snapshot=0
HADOOP: Job not successful!
HADOOP: Streaming Command Failed!
Job failed with return code 256: ['/diskb/dxb/code/hadoop-2.7.1/bin/hadoop', 'jar', '/diskb/dxb/code/hadoop-2.7.1/share/hadoop/tools/lib/hadoop-streaming-2.7.1.jar', '-files', 'hdfs:///user/root/tmp/mrjob/test.root.20150723.011910.649661/files/test.py#test.py,hdfs:///user/root/tmp/mrjob/test.root.20150723.011910.649661/files/setup-wrapper.sh#setup-wrapper.sh', '-archives', 'hdfs:///user/root/tmp/mrjob/test.root.20150723.011910.649661/files/mrjob.tar.gz#mrjob.tar.gz', '-input', 'hdfs:///user/root/tmp/mrjob/test.root.20150723.011910.649661/files/pg20417.txt', '-output', 'hdfs:///user/root/tmp/mrjob/test.root.20150723.011910.649661/output', '-mapper', 'sh -ex setup-wrapper.sh /root/.pyenv/versions/2.7.9/bin/python test.py --step-num=0 --mapper', '-combiner', 'sh -ex setup-wrapper.sh /root/.pyenv/versions/2.7.9/bin/python test.py --step-num=0 --combiner', '-reducer', 'sh -ex setup-wrapper.sh /root/.pyenv/versions/2.7.9/bin/python test.py --step-num=0 --reducer']
Scanning logs for probable cause of failure
Traceback (most recent call last):
  File "test.py", line 25, in <module>
    MRWordFreqCount.run()
  File "/root/.pyenv/versions/2.7.9/lib/python2.7/site-packages/mrjob/job.py", line 461, in run
    mr_job.execute()
  File "/root/.pyenv/versions/2.7.9/lib/python2.7/site-packages/mrjob/job.py", line 479, in execute
    super(MRJob, self).execute()
  File "/root/.pyenv/versions/2.7.9/lib/python2.7/site-packages/mrjob/launch.py", line 151, in execute
    self.run_job()
  File "/root/.pyenv/versions/2.7.9/lib/python2.7/site-packages/mrjob/launch.py", line 214, in run_job
    runner.run()
  File "/root/.pyenv/versions/2.7.9/lib/python2.7/site-packages/mrjob/runner.py", line 464, in run
    self._run()
  File "/root/.pyenv/versions/2.7.9/lib/python2.7/site-packages/mrjob/hadoop.py", line 237, in _run
    self._run_job_in_hadoop()
  File "/root/.pyenv/versions/2.7.9/lib/python2.7/site-packages/mrjob/hadoop.py", line 372, in _run_job_in_hadoop
    raise CalledProcessError(returncode, step_args)
subprocess.CalledProcessError: Command '['/diskb/dxb/code/hadoop-2.7.1/bin/hadoop', 'jar', '/diskb/dxb/code/hadoop-2.7.1/share/hadoop/tools/lib/hadoop-streaming-2.7.1.jar', '-files', 'hdfs:///user/root/tmp/mrjob/test.root.20150723.011910.649661/files/test.py#test.py,hdfs:///user/root/tmp/mrjob/test.root.20150723.011910.649661/files/setup-wrapper.sh#setup-wrapper.sh', '-archives', 'hdfs:///user/root/tmp/mrjob/test.root.20150723.011910.649661/files/mrjob.tar.gz#mrjob.tar.gz', '-input', 'hdfs:///user/root/tmp/mrjob/test.root.20150723.011910.649661/files/pg20417.txt', '-output', 'hdfs:///user/root/tmp/mrjob/test.root.20150723.011910.649661/output', '-mapper', 'sh -ex setup-wrapper.sh /root/.pyenv/versions/2.7.9/bin/python test.py --step-num=0 --mapper', '-combiner', 'sh -ex setup-wrapper.sh /root/.pyenv/versions/2.7.9/bin/python test.py --step-num=0 --combiner', '-reducer', 'sh -ex setup-wrapper.sh /root/.pyenv/versions/2.7.9/bin/python test.py --step-num=0 --reducer']' returned non-zero exit status 256
```

#4. the config in my hadoop are those:

##core-site.xml:
```javascript
core-site.xml:
<configuration>
    <property>
        <name>fs.defaultFS</name>
        <value>hdfs://master:9000</value>
    </property>
    <property>
      <name>hadoop.tmp.dir</name>
      <value>/tmp</value>
      <description>A base for other temporary directories.</description>
    </property>
</configuration>
```
##hdfs-site.xml:
```javascript
<configuration>
    <property>
        <name>dfs.replication</name>
        <value>1</value>
    </property>

    <property>
        <name>dfs.namenode.secondary.http-address</name>
        <value>master:9001</value>
    </property>
<property>
      <name>hadoop.tmp.dir</name>
      <value>/tmp</value>
      <description>A base for other temporary directories.</description>
    </property>
</configuration>
```
##mapred-site.xml:
```javascript
<configuration>
    <property>
        <name>mapreduce.framework.name</name>
        <value>yarn</value>
    </property>
    <property>
        <name>mapreduce.jobhistory.address</name>
        <value>master:10020</value>
    </property>

    <property>
        <name>mapreduce.jobhistory.webapp.address</name>
        <value>master:19888</value>
    </property>
</configuration>
```
##yarn-site.xml :
```javascript
<configuration>
    <property>
        <name>yarn.nodemanager.aux-services</name>
        <value>mapreduce_shuffle</value>
    </property>

    <property>                                                                
    <name>yarn.nodemanager.aux-services.mapreduce.shuffle.class</name>
        <value>org.apache.hadoop.mapred.ShuffleHandler</value>
    </property>
    <property>
        <name>yarn.resourcemanager.address</name>
        <value>master:8032</value>
    </property>
    <property>
        <name>yarn.resourcemanager.scheduler.address</name>
        <value>master:8030</value>
    </property>
    <property>
        <name>yarn.resourcemanager.resource-tracker.address</name>
        <value>master:8031</value>
    </property>
    <property>
        <name>yarn.resourcemanager.admin.address</name>
        <value>master:8033</value>
    </property>
    <property>
        <name>yarn.resourcemanager.webapp.address</name>
        <value>master:8088</value>
    </property>
<!-- Site specific YARN configuration properties -->
    <property>
        <name>yarn.scheduler.minimum-allocation-mb</name>
        <value>128</value>
        <description>Minimum limit of memory to allocate to each container request at the Resource Manager.</description>
    </property>
    <property>
        <name>yarn.scheduler.maximum-allocation-mb</name>
        <value>2048</value>
        <description>Maximum limit of memory to allocate to each container request at the Resource Manager.</description>
    </property>
    <property>
        <name>yarn.scheduler.minimum-allocation-vcores</name>
        <value>1</value>
        <description>The minimum allocation for every container request at the RM, in terms of virtual CPU cores. Requests lower than this won't take effect, and the specified value will get allocated the minimum.</description>
    </property>
    <property>
        <name>yarn.scheduler.maximum-allocation-vcores</name>
        <value>2</value>
        <description>The maximum allocation for every container request at the RM, in terms of virtual CPU cores. Requests higher than this won't take effect, and will get capped to this value.</description>
    </property>
    <property>
        <name>yarn.nodemanager.resource.memory-mb</name>
        <value>4096</value>
        <description>Physical memory, in MB, to be made available to running containers</description>
    </property>
    <property>
        <name>yarn.nodemanager.resource.cpu-vcores</name>
        <value>4</value>
        <description>Number of CPU cores that can be allocated for containers.</description>
    </property>
</configuration>
```

**Comment by davidmarin** [comment]
Do you still have stderr logs from the tasks that failed? That would tell you more. It could be something as simple as the wrong path for the Python binary.

**Comment by wsshopping** [comment]
hi davidmarin： thanks for you answers, in my stderr, it shows:
```
[root@node2 container_1437555855039_0012_01_000001]# cat  stderr 
log4j:WARN No appenders could be found for logger (org.apache.hadoop.ipc.Server).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
```
but when i use this command, nothing effect:
```
python test.py -r hadoop --interpreter=/root/.pyenv/versions/2.7.9/bin/python  ./pg20417.txt 
```
dose it matters the hadoop versions?  in my case， it is  hadoop:2.7.1

**Comment by wsshopping** [comment]
and when i  use  options  -v 
```
python test.py -r hadoop -v ./pg20417.txt  > output
```
it shows:
```
[root@master hadoop-2.7.1]# python test.py -r hadoop -v ./pg20417.txt  > output
looking for configs in /root/.mrjob.conf
looking for configs in /etc/mrjob.conf
using configs in /etc/mrjob.conf
Active configuration:
{'base_tmp_dir': '/tmp',
 'bootstrap_mrjob': True,
 'check_input_paths': True,
 'cleanup': ['ALL'],
 'cleanup_on_failure': ['NONE'],
 'cmdenv': {},
 'hadoop_bin': ['/diskb/dxb/code/hadoop-2.7.1/bin/hadoop'],
 'hadoop_extra_args': [],
 'hadoop_home': '/diskb/dxb/code/hadoop-2.7.1',
 'hadoop_streaming_jar': None,
 'hadoop_version': '0.20',
 'hdfs_scratch_dir': 'tmp/mrjob',
 'interpreter': ['/root/.pyenv/versions/2.7.9/bin/python'],
 'job_name': None,
 'jobconf': {},
 'label': None,
 'owner': 'root',
 'python_archives': [],
 'python_bin': ['python'],
 'setup': [],
 'setup_cmds': [],
 'setup_scripts': [],
 'sh_bin': ['sh', '-ex'],
 'steps_interpreter': ['/root/.pyenv/versions/2.7.9/bin/python'],
 'steps_python_bin': ['/root/.pyenv/versions/2.7.9/bin/python'],
 'strict_protocols': None,
 'upload_archives': [],
 'upload_files': []}
Looking for hadoop streaming jar in /diskb/dxb/code/hadoop-2.7.1
Hadoop streaming jar is /diskb/dxb/code/hadoop-2.7.1/share/hadoop/tools/lib/hadoop-streaming-2.7.1.jar
creating tmp directory /tmp/test.root.20150727.031714.703094
archiving /root/.pyenv/versions/2.7.9/lib/python2.7/site-packages/mrjob -> /tmp/test.root.20150727.031714.703094/mrjob.tar.gz as mrjob/
writing wrapper script to /tmp/test.root.20150727.031714.703094/setup-wrapper.sh
WRAPPER: # store $PWD
WRAPPER: __mrjob_PWD=$PWD
WRAPPER: 
WRAPPER: # obtain exclusive file lock
WRAPPER: exec 9>/tmp/wrapper.lock.test.root.20150727.031714.703094
WRAPPER: python -c 'import fcntl; fcntl.flock(9, fcntl.LOCK_EX)'
WRAPPER: 
WRAPPER: # setup commands
WRAPPER: {
WRAPPER:   export PYTHONPATH=$__mrjob_PWD/mrjob.tar.gz:$PYTHONPATH
WRAPPER: } 0</dev/null 1>&2
WRAPPER: 
WRAPPER: # release exclusive file lock
WRAPPER: exec 9>&-
WRAPPER: 
WRAPPER: # run task from the original working directory
WRAPPER: cd $__mrjob_PWD
WRAPPER: "$@"
Making directory hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/ on HDFS
> /diskb/dxb/code/hadoop-2.7.1/bin/hadoop version
Using Hadoop version 2.7.1
> /diskb/dxb/code/hadoop-2.7.1/bin/hadoop fs -mkdir -p hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/
Copying local files into hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/
Uploading ./pg20417.txt -> hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/pg20417.txt on HDFS
> /diskb/dxb/code/hadoop-2.7.1/bin/hadoop fs -put ./pg20417.txt hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/pg20417.txt
Uploading /diskb/dxb/code/hadoop-2.7.1/test.py -> hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/test.py on HDFS
> /diskb/dxb/code/hadoop-2.7.1/bin/hadoop fs -put /diskb/dxb/code/hadoop-2.7.1/test.py hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/test.py
Uploading /tmp/test.root.20150727.031714.703094/setup-wrapper.sh -> hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/setup-wrapper.sh on HDFS
> /diskb/dxb/code/hadoop-2.7.1/bin/hadoop fs -put /tmp/test.root.20150727.031714.703094/setup-wrapper.sh hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/setup-wrapper.sh
Uploading /tmp/test.root.20150727.031714.703094/mrjob.tar.gz -> hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/mrjob.tar.gz on HDFS
> /diskb/dxb/code/hadoop-2.7.1/bin/hadoop fs -put /tmp/test.root.20150727.031714.703094/mrjob.tar.gz hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/mrjob.tar.gz
> /root/.pyenv/versions/2.7.9/bin/python /diskb/dxb/code/hadoop-2.7.1/test.py --steps
running step 1 of 1
> /diskb/dxb/code/hadoop-2.7.1/bin/hadoop jar /diskb/dxb/code/hadoop-2.7.1/share/hadoop/tools/lib/hadoop-streaming-2.7.1.jar -files 'hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/test.py#test.py,hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/setup-wrapper.sh#setup-wrapper.sh' -archives 'hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/mrjob.tar.gz#mrjob.tar.gz' -input hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/pg20417.txt -output hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/output -mapper 'sh -ex setup-wrapper.sh /root/.pyenv/versions/2.7.9/bin/python test.py --step-num=0 --mapper' -combiner 'sh -ex setup-wrapper.sh /root/.pyenv/versions/2.7.9/bin/python test.py --step-num=0 --combiner' -reducer 'sh -ex setup-wrapper.sh /root/.pyenv/versions/2.7.9/bin/python test.py --step-num=0 --reducer'
HADOOP: packageJobJar: [/tmp/hadoop-unjar4465312523117728891/] [] /tmp/streamjob4994401619229872585.jar tmpDir=null
HADOOP: Error: java.lang.RuntimeException: PipeMapRed.waitOutputThreads(): subprocess failed with code 2
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRed.waitOutputThreads(PipeMapRed.java:322)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRed.mapRedFinished(PipeMapRed.java:535)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapper.close(PipeMapper.java:130)
HADOOP: 	at org.apache.hadoop.mapred.MapRunner.run(MapRunner.java:61)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRunner.run(PipeMapRunner.java:34)
HADOOP: 	at org.apache.hadoop.mapred.MapTask.runOldMapper(MapTask.java:453)
HADOOP: 	at org.apache.hadoop.mapred.MapTask.run(MapTask.java:343)
HADOOP: 	at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:164)
HADOOP: 	at java.security.AccessController.doPrivileged(Native Method)
HADOOP: 	at javax.security.auth.Subject.doAs(Subject.java:415)
HADOOP: 	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1657)
HADOOP: 	at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:158)
HADOOP: 
HADOOP: Error: java.lang.RuntimeException: PipeMapRed.waitOutputThreads(): subprocess failed with code 2
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRed.waitOutputThreads(PipeMapRed.java:322)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRed.mapRedFinished(PipeMapRed.java:535)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapper.close(PipeMapper.java:130)
HADOOP: 	at org.apache.hadoop.mapred.MapRunner.run(MapRunner.java:61)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRunner.run(PipeMapRunner.java:34)
HADOOP: 	at org.apache.hadoop.mapred.MapTask.runOldMapper(MapTask.java:453)
HADOOP: 	at org.apache.hadoop.mapred.MapTask.run(MapTask.java:343)
HADOOP: 	at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:164)
HADOOP: 	at java.security.AccessController.doPrivileged(Native Method)
HADOOP: 	at javax.security.auth.Subject.doAs(Subject.java:415)
HADOOP: 	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1657)
HADOOP: 	at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:158)
HADOOP: 
HADOOP: Error: java.lang.RuntimeException: PipeMapRed.waitOutputThreads(): subprocess failed with code 2
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRed.waitOutputThreads(PipeMapRed.java:322)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRed.mapRedFinished(PipeMapRed.java:535)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapper.close(PipeMapper.java:130)
HADOOP: 	at org.apache.hadoop.mapred.MapRunner.run(MapRunner.java:61)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRunner.run(PipeMapRunner.java:34)
HADOOP: 	at org.apache.hadoop.mapred.MapTask.runOldMapper(MapTask.java:453)
HADOOP: 	at org.apache.hadoop.mapred.MapTask.run(MapTask.java:343)
HADOOP: 	at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:164)
HADOOP: 	at java.security.AccessController.doPrivileged(Native Method)
HADOOP: 	at javax.security.auth.Subject.doAs(Subject.java:415)
HADOOP: 	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1657)
HADOOP: 	at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:158)
HADOOP: 
HADOOP: Error: java.lang.RuntimeException: PipeMapRed.waitOutputThreads(): subprocess failed with code 2
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRed.waitOutputThreads(PipeMapRed.java:322)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRed.mapRedFinished(PipeMapRed.java:535)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapper.close(PipeMapper.java:130)
HADOOP: 	at org.apache.hadoop.mapred.MapRunner.run(MapRunner.java:61)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRunner.run(PipeMapRunner.java:34)
HADOOP: 	at org.apache.hadoop.mapred.MapTask.runOldMapper(MapTask.java:453)
HADOOP: 	at org.apache.hadoop.mapred.MapTask.run(MapTask.java:343)
HADOOP: 	at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:164)
HADOOP: 	at java.security.AccessController.doPrivileged(Native Method)
HADOOP: 	at javax.security.auth.Subject.doAs(Subject.java:415)
HADOOP: 	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1657)
HADOOP: 	at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:158)
HADOOP: 
HADOOP: Error: java.lang.RuntimeException: PipeMapRed.waitOutputThreads(): subprocess failed with code 2
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRed.waitOutputThreads(PipeMapRed.java:322)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRed.mapRedFinished(PipeMapRed.java:535)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapper.close(PipeMapper.java:130)
HADOOP: 	at org.apache.hadoop.mapred.MapRunner.run(MapRunner.java:61)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRunner.run(PipeMapRunner.java:34)
HADOOP: 	at org.apache.hadoop.mapred.MapTask.runOldMapper(MapTask.java:453)
HADOOP: 	at org.apache.hadoop.mapred.MapTask.run(MapTask.java:343)
HADOOP: 	at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:164)
HADOOP: 	at java.security.AccessController.doPrivileged(Native Method)
HADOOP: 	at javax.security.auth.Subject.doAs(Subject.java:415)
HADOOP: 	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1657)
HADOOP: 	at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:158)
HADOOP: 
HADOOP: Error: java.lang.RuntimeException: PipeMapRed.waitOutputThreads(): subprocess failed with code 2
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRed.waitOutputThreads(PipeMapRed.java:322)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRed.mapRedFinished(PipeMapRed.java:535)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapper.close(PipeMapper.java:130)
HADOOP: 	at org.apache.hadoop.mapred.MapRunner.run(MapRunner.java:61)
HADOOP: 	at org.apache.hadoop.streaming.PipeMapRunner.run(PipeMapRunner.java:34)
HADOOP: 	at org.apache.hadoop.mapred.MapTask.runOldMapper(MapTask.java:453)
HADOOP: 	at org.apache.hadoop.mapred.MapTask.run(MapTask.java:343)
HADOOP: 	at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:164)
HADOOP: 	at java.security.AccessController.doPrivileged(Native Method)
HADOOP: 	at javax.security.auth.Subject.doAs(Subject.java:415)
HADOOP: 	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1657)
HADOOP: 	at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:158)
HADOOP: 
HADOOP: Streaming Command Failed!
Job failed with return code 256: ['/diskb/dxb/code/hadoop-2.7.1/bin/hadoop', 'jar', '/diskb/dxb/code/hadoop-2.7.1/share/hadoop/tools/lib/hadoop-streaming-2.7.1.jar', '-files', 'hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/test.py#test.py,hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/setup-wrapper.sh#setup-wrapper.sh', '-archives', 'hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/mrjob.tar.gz#mrjob.tar.gz', '-input', 'hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/pg20417.txt', '-output', 'hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/output', '-mapper', 'sh -ex setup-wrapper.sh /root/.pyenv/versions/2.7.9/bin/python test.py --step-num=0 --mapper', '-combiner', 'sh -ex setup-wrapper.sh /root/.pyenv/versions/2.7.9/bin/python test.py --step-num=0 --combiner', '-reducer', 'sh -ex setup-wrapper.sh /root/.pyenv/versions/2.7.9/bin/python test.py --step-num=0 --reducer']
Scanning logs for probable cause of failure
Traceback (most recent call last):
  File "test.py", line 23, in <module>
    MRWordFreqCount.run()
  File "/root/.pyenv/versions/2.7.9/lib/python2.7/site-packages/mrjob/job.py", line 461, in run
    mr_job.execute()
  File "/root/.pyenv/versions/2.7.9/lib/python2.7/site-packages/mrjob/job.py", line 479, in execute
    super(MRJob, self).execute()
  File "/root/.pyenv/versions/2.7.9/lib/python2.7/site-packages/mrjob/launch.py", line 151, in execute
    self.run_job()
  File "/root/.pyenv/versions/2.7.9/lib/python2.7/site-packages/mrjob/launch.py", line 214, in run_job
    runner.run()
  File "/root/.pyenv/versions/2.7.9/lib/python2.7/site-packages/mrjob/runner.py", line 464, in run
    self._run()
  File "/root/.pyenv/versions/2.7.9/lib/python2.7/site-packages/mrjob/hadoop.py", line 237, in _run
    self._run_job_in_hadoop()
  File "/root/.pyenv/versions/2.7.9/lib/python2.7/site-packages/mrjob/hadoop.py", line 372, in _run_job_in_hadoop
    raise CalledProcessError(returncode, step_args)
subprocess.CalledProcessError: Command '['/diskb/dxb/code/hadoop-2.7.1/bin/hadoop', 'jar', '/diskb/dxb/code/hadoop-2.7.1/share/hadoop/tools/lib/hadoop-streaming-2.7.1.jar', '-files', 'hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/test.py#test.py,hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/setup-wrapper.sh#setup-wrapper.sh', '-archives', 'hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/mrjob.tar.gz#mrjob.tar.gz', '-input', 'hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/files/pg20417.txt', '-output', 'hdfs:///user/root/tmp/mrjob/test.root.20150727.031714.703094/output', '-mapper', 'sh -ex setup-wrapper.sh /root/.pyenv/versions/2.7.9/bin/python test.py --step-num=0 --mapper', '-combiner', 'sh -ex setup-wrapper.sh /root/.pyenv/versions/2.7.9/bin/python test.py --step-num=0 --combiner', '-reducer', 'sh -ex setup-wrapper.sh /root/.pyenv/versions/2.7.9/bin/python test.py --step-num=0 --reducer']' returned non-zero exit status 256
```

**Comment by tommyjcarpenter** [comment]
I am seeing this exact same error.

**Comment by davidmarin** [comment]
Looks like I'm going to need to set up Hadoop and try to duplicate this. Thanks!

**Comment by davidmarin** [comment]
This sounds like an issue with your shell wrapper script, which may have been fixed by #1078.

I'm a little confused as to why "python" and not your custom `python_bin` appears in the setup wrapper script.

**Comment by davidmarin** [comment]
Guessing this was fixed by #1078. This is really a wrapper script issue, not a Hadoop issue.
